app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -370,6 +350,45 @@ public void handleOnBackPressed() {
         });
     }
 
+    private SeekBar.OnSeekBarChangeListener setOnSeekBarChangeListenerUnifiedFunction() {
+        return new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                if (!fromUser) {
+                    Log.d(TAG, "non user triggered onProgressChanged, ignoring, progress is " + progress);
+                    return;
+                }
+                Log.d(TAG, "Progress is " + progress);
+                if (seekBar.getId() == binding.barcodeScaler.getId()) {
+                    Log.d(TAG, "Max is " + binding.barcodeScaler.getMax());
+                    float scale = (float) progress / (float) binding.barcodeScaler.getMax();
+                    Log.d(TAG, "Scaling to " + scale);
+                }
+                else {
+                    Log.d(TAG, "Max is " + binding.barcodeWidthscaler.getMax());
+                    float scale = (float) progress / (float) binding.barcodeWidthscaler.getMax();
+                    Log.d(TAG, "Scaling to " + scale);
+                }
+                DBHelper.updateLoyaltyCardZoomLevel(database, loyaltyCardId, loyaltyCard.zoomLevel, loyaltyCard.zoomLevelWidth);
+                if (seekBar.getId() == binding.barcodeScaler.getId()) {
+                    loyaltyCard.zoomLevel = progress;
+                    setScalerGuideline(loyaltyCard.zoomLevel);
+                }
+                else {
+                    loyaltyCard.zoomLevelWidth = progress;
+                    setScalerWidthGuideline(loyaltyCard.zoomLevelWidth);
+                }
+                drawMainImage(mainImageIndex, true, isFullscreen);
```

Comment:
I think this logic is the wrong way around and it just happens to work "well enough" because we tend to drag the bar for more than one value (but the saved value will always be one off).

(For safety, let's do the DB update last. If we somehow set a value that causes a crash it won't be saved to the DB)

```suggestion
                if (seekBar.getId() == binding.barcodeScaler.getId()) {
                    loyaltyCard.zoomLevel = progress;
                    setScalerGuideline(loyaltyCard.zoomLevel);
                }
                else {
                    loyaltyCard.zoomLevelWidth = progress;
                    setScalerWidthGuideline(loyaltyCard.zoomLevelWidth);
                }
                drawMainImage(mainImageIndex, true, isFullscreen);
                
                DBHelper.updateLoyaltyCardZoomLevel(database, loyaltyCardId, loyaltyCard.zoomLevel, loyaltyCard.zoomLevelWidth);
```

Reply:
Thanks for all your work, and specially for your reviews, I hope I learned a lot from you :)
I've just commited your suggestion in 1ed2b01c and rebased again.
HTH
Pere